UnchainedLauncher.Core/Services/Mods/Registry/LocalModRegistry.cs

```diff
@@ -26,6 +30,103 @@ public override EitherAsync<ModPakStreamAcquisitionFailure, FileWriter> Download
                 .Map(sizedStream => new FileWriter(outputLocation, sizedStream.Stream, sizedStream.Size));
         }
 
+        public async Task DeleteRelease(ReleaseCoordinates coordinates) {
+            var _ = await GetMod(coordinates)
+                .Match(
+                mod => {
+                    var deletedRelease = mod.Releases.Filter(r =>
+                        coordinates.Matches(r) && r.Tag.Equals(coordinates.Version)
+                    ).FirstOrDefault();
+
+                    if (deletedRelease == null) {
+                        return Unit.Default;
+                    }
+
+                    var deleted = mod with {
+                        Releases = mod.Releases
+                            .Filter(x => x != deletedRelease)
+                            .ToList()
+                    };
+
+                    var modPath = Path.Combine(RegistryPath, coordinates.Org, coordinates.ModuleName);
+                    var releasePath = Path.Combine(modPath, deletedRelease.Tag);
+                    if (!deleted.Releases.Any()) {
+                        Directory.Delete(modPath, true);
+                        return Unit.Default;
+                    }
+
+                    Directory.Delete(releasePath, true);
+
+                    WriteMod(deleted with {
+                        // we know this unwrap is safe because if there were no releases, we would have deleted
+                        LatestManifest = deleted.LatestRelease.ValueUnsafe().Manifest
+                    });
+                    return Unit.Default;
+                },
+                _ => Unit.Default
```

Comment:
We should probably log a warning when this happens. If there's an attempt to delete something that doesn't exist, that could indicate an issue with inconsistent state somewhere.